x/precisebank/types/fractional_balances.go

```diff
@@ -0,0 +1,42 @@
+package types
+
+import (
+	fmt "fmt"
+	"strings"
+)
+
+// FractionalBalances is a slice of FractionalBalance
+type FractionalBalances []FractionalBalance
+
+// Len returns the length of the slice.
+func (fbs FractionalBalances) Len() int {
+	return len(fbs)
+}
+
+// Validate returns an error if any FractionalBalance in the slice is invalid.
+func (fbs FractionalBalances) Validate() error {
+	seenAddresses := make(map[string]struct{})
+
+	for _, fb := range fbs {
+		// Ensure there is no white-space before/after address as that can
+		// bypass duplicate address check.
+		// Upper/lowercase is not an issue as the address is a bech32 encoded
+		// string which only allows lowercase.
```

Comment:
This comment conflicts with the test case the error message `decoding bech32 failed: string not all lowercase or all uppercase` in FractionalBalance validation test cases.